dist/scripts/test/src/MillVersionFrontmatterTests.scala

```diff
@@ -0,0 +1,88 @@
+package mill.scripts
+
+import utest.framework.TestPath
+import utest.*
+
+object MillVersionFrontmatterTests extends TestSuite {
```

Comment:
Let's test this using `MILL_TEST_DRY_RUN_LAUNCHER_SCRIPT=true`. That would make the tests run much faster than actually running Mill and bootstrapping a Mill daemon which is what the code does now, while still exercising the important part of the code which is the `mill.bat`/`mill.sh` parser logic